src/app/core/entity/entity-remove.service.spec.ts

```diff
@@ -47,22 +66,27 @@ describe("EntityRemoveService", () => {
   it("should return false when user cancels confirmation", async () => {
     mockConfirmationDialog.getConfirmation.and.resolveTo(false);
 
-    const result = await service.remove(new Entity());
+    const result = await service.delete(new Entity());
 
     expect(result).toBe(false);
     expect(snackBarSpy.open).not.toHaveBeenCalled();
-    expect(mockEntityMapper.remove).not.toHaveBeenCalled();
+    expect(mockedEntityMapper.remove).not.toHaveBeenCalled();
   });
 
   it("should delete entity, show snackbar confirmation and navigate back", async () => {
     // onAction is never called
     mockSnackBarRef.onAction.and.returnValues(NEVER);
+    // mock that dialog is dismissed immediately
+    const afterDismissed = new Observable<MatSnackBarDismiss>((subscriber) =>
+      subscriber.next({} as MatSnackBarDismiss),
+    );
+    mockSnackBarRef.afterDismissed.and.returnValue(afterDismissed);
```

Comment:
```suggestion
    mockSnackBarRef.afterDismissed.and.returnValue(of(undefined));
```
Or is there any reason for this more complex approach?

Reply:
have kept this from previous code - yours looks nice!